src/components/ClockFace.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from 'react';
+import classnames from 'classnames';
+import SettingsModal from './SettingsModal';
+
+const ClockFace = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isActive, setIsActive] = useState(false);
+  const [currentSessionIndex, setCurrentSessionIndex] = useState(0);
+  // in seconds
+  const [sessions, setSessions] = useState([60, 5, 10, 10]);
+  const [currentSeconds, setCurrentSeconds] = useState(sessions[currentSessionIndex]);
+
+  useEffect(() => {
+    let timeInterval;
+    if (isActive) {
+      timeInterval = setInterval(() => {
+        if (currentSeconds > 0) {
+          setCurrentSeconds(currentSeconds - 1);
+        } else {
+          setIsActive(false);
+          clearInterval(timeInterval);
+        }
+      }, 1000)
+    }
+    else if (!isActive) {
+      clearInterval(timeInterval)
+    }
+    return () => clearInterval(timeInterval)
+  }, [isActive, currentSeconds]);
+
+  useEffect(() => {
+    if (isActive && currentSeconds === 0) {
```

Comment:
This `useEffect` is using the exact dependencies as the previous ones and the logic are very similar. I feel like you can somehow consolidate the two functions' logic, otherwise it will be very hard to debug if anything goes wrong.